app/models/goal.py

```diff
@@ -1,5 +1,14 @@
 from app import db
-
+from flask import current_app
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", backref="task", lazy=True)
+
+    def make_dict(self):
```

Comment:
Nice helper, I would also suggest maybe making a helper method which takes a dict and creates a Goal from it.   The same with Task.